core/trino-parser/src/main/java/io/trino/sql/tree/BinaryLiteral.java

```diff
@@ -63,10 +61,10 @@ public BinaryLiteral(NodeLocation location, String value)
      */
     public String toHexString()
     {
-        return BaseEncoding.base16().encode(value.getBytes());
+        return BaseEncoding.base16().encode(value);
     }
 
-    public Slice getValue()
+    public byte[] getValue()
     {
         return value;
```

Comment:
Call `clone()` here so that callers can't mutate the value (and thus violate the immutability of the tree)